vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -405,6 +409,80 @@ private void sendFile(RoutingContext context, String file, FileProps fileProps)
             }
           }
 
+          // http2 pushing support
+          if (request.version() == HttpVersion.HTTP_2 && http2PushMapping != null) {
```

Comment:
Now I'm not sure if this is the right place to handle preload. By doing it on the static handler it means that only hard coded at start up time static files can be preloaded and I guess this would be a dynamic action so it probably makes more sense to live under the `RoutingContext` object.

Reply:
I didn't get that part, could you explain a bit more about how I can do it under a RoutingContext object.